tfsdk/plan_test.go

```diff
@@ -389,32 +821,2518 @@ func TestPlanSetAttribute(t *testing.T) {
 	}
 
 	testCases := map[string]testCase{
-		"basic": {
+		"add-Bool": {
 			plan: Plan{
 				Raw: tftypes.NewValue(tftypes.Object{
 					AttributeTypes: map[string]tftypes.Type{
-						"name": tftypes.String,
+						"other": tftypes.String,
+					},
+				}, map[string]tftypes.Value{
+					"other": tftypes.NewValue(tftypes.String, "should be untouched"),
+				}),
+				Schema: Schema{
+					Attributes: map[string]Attribute{
+						"test": {
+							Type:     types.BoolType,
+							Required: true,
+						},
+						"other": {
+							Type:     types.StringType,
+							Required: true,
+						},
+					},
+				},
+			},
+			path: tftypes.NewAttributePath().WithAttributeName("test"),
+			val:  false,
+			expected: tftypes.NewValue(tftypes.Object{
+				AttributeTypes: map[string]tftypes.Type{
+					"test":  tftypes.Bool,
+					"other": tftypes.String,
+				},
+			}, map[string]tftypes.Value{
+				"test":  tftypes.NewValue(tftypes.Bool, false),
+				"other": tftypes.NewValue(tftypes.String, "should be untouched"),
+			}),
+		},
+		"add-List": {
+			plan: Plan{
+				Raw: tftypes.NewValue(tftypes.Object{
+					AttributeTypes: map[string]tftypes.Type{
+						"other": tftypes.String,
+					},
+				}, map[string]tftypes.Value{
+					"other": tftypes.NewValue(tftypes.String, "should be untouched"),
+				}),
+				Schema: Schema{
+					Attributes: map[string]Attribute{
+						"tags": {
```

Comment:
Same, these attributes would be in the Plan's `Raw`, set to `null`, not omitted from it.